### ISSUE TITLE
false positive double_negations warning with macro invocation

### ISSUE THREAD

**Issue by anatawa12** [rust-lang/rust#143980]
I tried this code:

```rust

macro_rules! neg {
    ($value: expr) => { -$value }
}

fn test() {
    let value = 1;
    let _ = neg!(-value);
}

```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=0e6ac86ed0b4b3573551f8537f083c7a

I expected to see this happen: No double_negations warnings

Instead, this happened: double_negations warning

### Meta

`rustc --version --verbose`: 
```
1.90.0-nightly
(2025-07-14 a001497644bc229f1abc)
```
(Playgournd)
